src/Void.Core/CommandMode.fs

```diff
@@ -8,13 +8,14 @@ module CommandMode =
         | EntryCancelled
         | CharacterBackspaced
         | TextAppended of string
-        | CommandCompleted
+        | TextReplaced of string
+        | CommandCompleted of string
         interface EventMessage
 
     let private handleEnter interpret buffer =
         match interpret { Language = "VoidScript"; Fragment = buffer} with
         | InterpretScriptFragmentResponse.Completed ->
-            ("", Event.CommandCompleted :> Message)
+            ("", Event.CommandCompleted buffer :> Message)
         | InterpretScriptFragmentResponse.ParseFailed error ->
             ("", CoreEvent.ErrorOccurred error :> Message)

```

Comment:
@mjgpy3 For the error, seems like you could add the command that failed as information to the error type that says that command failed, and then in your command history add a core event handler that pattern matches for `ErrorOccurred` only for `ParseFailed` errors. Or something to that effect... the fact that the error is already pre-built, not build in the command mode service, might make things a little messier. Might be in for a bit of a refactor on that.